pkg/networkservice/chains/endpoint/server.go

```diff
@@ -61,13 +61,12 @@ type endpoint struct {
 //             - additionalFunctionality - any additional NetworkServiceServer chain elements to be included in the chain
 func NewServer(ctx context.Context, name string, authzServer networkservice.NetworkServiceServer, tokenGenerator token.GeneratorFunc, additionalFunctionality ...networkservice.NetworkServiceServer) Endpoint {
 	rv := &endpoint{}
-	var ns networkservice.NetworkServiceServer = rv
 	rv.NetworkServiceServer = chain.NewNetworkServiceServer(
 		append([]networkservice.NetworkServiceServer{
 			authzServer,
 			updatepath.NewServer(name),
+			timeout.NewServer(ctx),
```

Comment:
This is a big improvement in ordering :)

Reply:
It is really important because now `timeout` closes only subsequent chain elements and `monitor` needs to be closed.

Reply:
Totally agree.

Reply:
I also very like this direction. It looks better than the first implementation that I saw in this PR based on an additional chain break element.